File: src/main/kotlin/io/mcm/springbootwithkotlin/model/BooksResponse.java
package io.mcm.springbootwithkotlin.model;

import io.mcm.springbootwithkotlin.model.entities.Book;

import java.util.List;

public class BooksResponse {
    List<Book> bookList;
    private int totalPages;
    private long totalElements;
    private int pageNo;

    public BooksResponse(List<Book> bookList) {
        this.bookList = bookList;
    }

    @Override
    public String toString() {
        return "BooksResponse{" +
                "bookList=" + bookList +
                '}';
    }

    public List<Book> getBookList() {
        return bookList;
    }

    public void setBookList(List<Book> bookList) {
        this.bookList = bookList;
    }

    public void setTotalPages(int totalPages) {
        this.totalPages = totalPages;
    }

    public void setTotalCount(long totalElements) {
        this.totalElements = totalElements;
    }

    public void setPageNo(int i) {
        this.pageNo = i;
    }
}

File: src/main/kotlin/io/mcm/springbootwithkotlin/model/BooksResponse.kt
package io.mcm.springbootwithkotlin.model

import io.mcm.springbootwithkotlin.model.entities.Book

class BooksResponse(var bookList: MutableList<Book?>) {
    private var totalPages = 0
    private var totalElements: Long = 0
    private var pageNo = 0
    override fun toString(): String {
        return "BooksResponse{" +
                "bookList=" + bookList +
                '}'
    }

    fun setTotalPages(totalPages: Int) {
        this.totalPages = totalPages
    }

    fun setTotalCount(totalElements: Long) {
        this.totalElements = totalElements
    }

    fun setPageNo(i: Int) {
        pageNo = i
    }
}